phoenix-core/src/main/java/org/apache/phoenix/schema/json/PhoenixJson.java

```diff
@@ -0,0 +1,355 @@
+            // ignore
+        }
+        return null;
+    }
+
+    /**
+     * Serialize the current {@link PhoenixJson} to String. Its required for
+     * json_extract_path_text(). If we just return node.toString() it will wrap String value in
+     * double quote which is not the expectation, hence avoiding calling toString() on
+     * {@link JsonNode} until PhoenixJson represent a Json Array or container for Json object. If
+     * PhoenixJson just represent a {@link ValueNode} then it should return value returned from
+     * objects toString().
+     */
+    public String serializeToString() {
+        if (this.rootNode == null || this.rootNode.isNull()) {
+            return null;
+        } else if (this.rootNode.isValueNode()) {
+
+            if (this.rootNode.isNumber()) {
+                return this.rootNode.getNumberValue().toString();
+            } else if (this.rootNode.isBoolean()) {
+                return String.valueOf(this.rootNode.getBooleanValue());
+            } else if (this.rootNode.isTextual()) {
+                return this.rootNode.getTextValue();
+            } else {
+                return this.jsonAsString;
+            }
+        } else if (this.rootNode.isArray()) {
+            return this.jsonAsString;
+        } else if (this.rootNode.isContainerNode()) {
+            return this.jsonAsString;
+        }
+
+        return null;
+
+    }
+
+    @Override
+    public String toString() {
+        return this.jsonAsString;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + this.jsonAsString.hashCode();
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        throw new EqualityNotSupportedException(PJson.INSTANCE);
+    }
+
+    /**
+     * @return length of the string represented by the current {@link PhoenixJson}.
+     */
+    public int estimateByteSize() {
+        String jsonStr = toString();
+        return jsonStr == null ? 1 : jsonStr.length();
+    }
+
+    public byte[] toBytes() {
+        return Bytes.toBytes(this.jsonAsString);
+    }
+
+    @Override
+    public int compareTo(PhoenixJson o) {
+        throw new EqualityNotSupportedException(PJson.INSTANCE);
+    }
+
+    private PhoenixJson getPhoenixJsonInternal(String[] paths) {
+        JsonNode node = this.rootNode;
+        for (String path : paths) {
+            JsonNode nodeTemp = null;
+            if (node.isArray()) {
+                int index = Integer.parseInt(path);
+                nodeTemp = node.path(index);
+            } else {
+                nodeTemp = node.path(path);
+            }
+            if (nodeTemp == null || nodeTemp.isMissingNode()) {
+                return null;
+            }
+            node = nodeTemp;
+        }
+        return new PhoenixJson(node, node.toString());
+    }
+
+    public int getJsonArrayLength() {
+        int count = 0;
```

Comment:
Please add comment here, its hard to follow why we require this method and where we gonna use this. 

Probably it seems it will be used for json_array_length() function implementation. In that case first requirement will be to make sure that input is a valid Json array and if it is so then just check whether the rootnode (member variable of this class) isArray() or not, if it is an array then simply use size() method of ArrayNode class.